### DISCUSSION TITLE
Turn lights on at 6.30, but check for sunrise-10 min

### DISCUSSION THREAD

**Discussion by Reinier1999** [arendst/Tasmota#16647] [Q&A] [answered]
Hi! 
I have a question. I'm kind of a beginner with programming and Github too, so please correct me I I'm doing something wrong :)

I want to turn the lights on in the morning at 6.30, but turn them off 10 minutes before sunrise.
For turning on I found the following code:

Rule1
  ON Time#Minute=390 DO Backlog event checksunrise=%time% ENDON
  ON event#checksunrise<%sunrise% DO Power1 1 ENDON

Witch I modified to:

Rule1
  ON Time#Minute=390 DO Backlog event checksunrise=%time% ENDON
  ON event#checksunrise<%sunrise%-12 DO Power1 1 ENDON
(If 6.30 (390 minutes after midnight) = more than 12 minutes before sunrise, don't turn on)(so the light will stay off if it would run for less than two minutes in the morning) 

But this does not seem to work, it seems to ignore the -12 part totally. I suspect a var/mem issue, but can't get it to work.
Can anyone help me with this?

Thanks in advance!

Regards,
Reinier

**Comment by sfromis** [comment, marked as answer]
The [Timers](https://tasmota.github.io/docs/Timers/) feature (available in the web UI, Configuration/Timers) includes support for timing relative to sunrise/sunset, and also fixed timing. Should be pretty simple for your plan.

You cannot have arithmetic in the trigger part of a rule. That is a feature invented by you, but not existing in the docs.
